client/src/docs-ui/select-anchor/reroute-if-necessary.worker.ts

```diff
@@ -0,0 +1,21 @@
+import {parseURL} from "../../utils/url/url.worker";
+
+const rereouteCache = new Map<string, string>();
+
+export const rerouteIfNecessary = async (path: string) => {
+  if (rereouteCache.has(path)) {
+    return rereouteCache.get(path);
+  }
+  const rerouted = await (async () => {
+    if (
+      path.includes("/lib") &&
+      (path.includes("/q/platform/ios") || path.includes("/q/platform/android"))
+    ) {
+      return `/lib/q/platform/${(await parseURL(path)).params
```

Comment:
What happens with the path param fails here? Does it break us?

Reply:
What do you mean?

Reply:
I think he is asking what happens when the underlying `serializeURL` fails

Reply:
It won't (see how it ensures the path includes the virtual query params)